### DISCUSSION TITLE
Maximum speed?

### DISCUSSION THREAD

**Discussion by tpwrules** [svpcom/wfb-ng#326] [Questions] [answered]
I cannot get more than about 65Mbit/s transmit using the Alfa AWUS036ACH and using raw `wfb_tx` and iperf2 in UDP mode. I set MCS7, short guard band, 40MHz, channel 36, and packet size 1440 bytes. I also disabled FEC completely (set k=8 and n=8). According to the MCS index table, the raw data rate is 150Mbit/s and I was hoping to get closer to this. If I try MCS=4 with the other settings the same, I get about 50Mbit/s which is higher than I would expect.

Is this a fundamental limitation of WiFi? Is there some listen-before-talk mode or something to turn off? I tried some other cards and versions of the driver and couldn't do better, but I don't know where to look for the problem.

**Comment by svpcom** [comment]
![изображение](https://github.com/svpcom/wfb-ng/assets/503830/9ff6d360-cbb2-4123-87b5-245b1f9692e0)
speed in the table is **raw channel speed**. It doesn't include packet preamble and guard interval. So effective speed is ~ 1/2 of this speed

**Reply by tpwrules** [reply]
I had expected the table numbers to include the guard interval as the numbers are different for different guard intervals.

I’m also a little confused how I get 43% of the table speed at MCS7 and 55% at MCS4.

How long does the preamble take to transmit? I assume it’s the same independent of MCS.

**Reply by svpcom** [reply]
Yes, preamble speed is independent from mcs and it is very very low (to be compatible with 802.11b/g)

**Reply by svpcom** [reply, marked as answer]
See pdfs in doc directory

**Reply by tpwrules** [reply]
Thank you for the reference, it is very illuminating, though a bit disappointing.

Is it possible to turn up the MPDU size? It’s not immediately clear to me why wfb-ng is limited to 1445 bytes of user data and can’t take advantage of a 2 or 3KB MPDU like evaluated in the paper.

**Reply by svpcom** [reply]
You can use larger mtu, but not all drivers (ath9k, etc) supports it properly. With very large mtu 8812au sometimes trigger kernel panic

**Reply by svpcom** [reply]
So wfb-ng use standrard 1500 mtu minus packet header size. See define in wifibroadcast.hpp

**Reply by tpwrules** [reply]
Thanks for the pointer, looks like I can change it. I might investigate the kernel panic.
